Mathlib/Tactic/Linter/DocPrime.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Lean.Elab.Command
+
+/-!
+#  The "docPrime" linter
+
+The "docPrime" linter emits a warning on declarations that have no doc-string and whose
+name ends with a `'`.
+-/
+
+open Lean Elab
+
+namespace Mathlib.Linter
+
+/--
+The "docPrime" linter emits a warning on declarations that have no doc-string and whose
+name ends with a `'`.
+
+The file `scripts/no_lints_prime_decls.txt` records the list of exceptions at the moment in
+which the linter was written.
+-/
+register_option linter.docPrime : Bool := {
+  defValue := false
+  descr := "enable the docPrime linter"
+}
+
+namespace DocPrime
+
+@[inherit_doc Mathlib.Linter.linter.docPrime]
+def docPrimeLinter : Linter where run := withSetOptionIn fun stx ↦ do
+  unless Linter.getLinterValue linter.docPrime (← getOptions) do
+    return
+  if (← get).messages.hasErrors then
+    return
+  unless [``Lean.Parser.Command.declaration, `lemma].contains stx.getKind  do return
+  let docstring := stx[0][0]
+  let declId :=
```

Comment:
I just pushed a commit clarifying/commenting this slightly.

That said, this kind of syntax parsing feels quite brittle. I wonder if there's a better way to do this... but perhaps this is fine as-is.

Reply:
To clarify: I won't block landing the linter on that. It's worth thinking about, but not worth stalling this PR on.

Reply:
An alternative could be to retrieve the declaration name and ask for `hasDocstring` or similar.  In any case, I would probably worry about this if we get reports of the linter not behaving as expected.